iceoryx_posh/test/moduletests/test_mepoo_config.cpp

```diff
@@ -0,0 +1,153 @@
+    constexpr uint32_t chunkCount{100U};
+
+    sut.addMemPool({size, chunkCount});
+
+    EXPECT_EQ(sut.m_mempoolConfig.size(), 1U);
+    EXPECT_EQ(sut.m_mempoolConfig[0].m_size, size);
+    EXPECT_EQ(sut.m_mempoolConfig[0].m_chunkCount, chunkCount);
+}
+
+TEST_F(MePooConfig_Test, AddingMempoolWhenTheMemPoolConfigContainerIsFullReturnsError)
+{
+    MePooConfig sut;
+    constexpr uint32_t size{128U};
+    constexpr uint32_t chunkCount{100U};
+
+    for (size_t i = 0; i < iox::MAX_NUMBER_OF_MEMPOOLS; i++)
+    {
+        sut.addMemPool({size, chunkCount});
+    }
+    EXPECT_DEATH({ sut.addMemPool({size, chunkCount}); }, ".*");
+}
+
+TEST_F(MePooConfig_Test, SetDefaultMethodAddsTheDefaultMemPoolConfigurationToTheMemPoolConfigContainer)
+{
+    MePooConfig sut;
+    MePooConfig::Entry defaultEntry[7] = {{128U, 10000U},
+                                          {1024U, 5000U},
+                                          {1024U * 16U, 1000U},
+                                          {1024U * 128U, 200U},
+                                          {1024U * 512U, 50U},
+                                          {1024 * 1024, 30U},
+                                          {1024U * 1024U * 4U, 10U}};
+
+    sut.setDefaults();
+
+    EXPECT_EQ(sut.m_mempoolConfig.size(), 7U);
+    for (size_t i = 0; i < 7; i++)
+    {
+        EXPECT_EQ((sut.m_mempoolConfig[i].m_chunkCount), defaultEntry[i].m_chunkCount);
+        EXPECT_EQ((sut.m_mempoolConfig[i].m_size), defaultEntry[i].m_size);
+    }
+}
+
+TEST_F(MePooConfig_Test, GetMemoryConfigMethodReturnsTheMemPoolConfigContainerWithAddedMempools)
+{
+    MePooConfig sut;
+    constexpr uint32_t chunkCount{100U};
+    constexpr uint32_t size{128U};
+    sut.addMemPool({size, chunkCount});
+
+    const MePooConfig::MePooConfigContainerType* mempoolconfptr = sut.getMemPoolConfig();
+
+    EXPECT_THAT(mempoolconfptr->size(), Eq(1U));
+    EXPECT_THAT((mempoolconfptr[0].data()->m_chunkCount), Eq(chunkCount));
+    EXPECT_THAT((mempoolconfptr[0].data()->m_size), Eq(size));
+}
+
+TEST_F(MePooConfig_Test, OptimizeMethodCombinesTwoMempoolWithSameSizeAndDoublesTheChunkCountInTheMemPoolConfigContainer)
+{
+    MePooConfig sut;
+    constexpr uint32_t chunkCount{100U};
+    constexpr uint32_t size{100U};
+    sut.addMemPool({size, chunkCount});
+    sut.addMemPool({size, chunkCount});
+
+    sut.optimize();
+
+    EXPECT_THAT(sut.m_mempoolConfig.size(), Eq(1U));
+    EXPECT_THAT(sut.m_mempoolConfig[0].m_chunkCount, Eq(chunkCount * 2U));
+}
+
+TEST_F(MePooConfig_Test, OptimizeMethodRemovesTheMempoolWithSizeZeroInTheMemPoolConfigContainer)
+{
+    MePooConfig sut;
+    constexpr uint32_t chunkCount{100U};
+    constexpr uint32_t size1{64U};
+    constexpr uint32_t size2{0U};
+    constexpr uint32_t size3{128U};
+    sut.addMemPool({size1, chunkCount});
+    sut.addMemPool({size2, chunkCount});
+    sut.addMemPool({size3, chunkCount});
+
+    sut.optimize();
+
+    EXPECT_THAT(sut.m_mempoolConfig.size(), Eq(2U));
```

Comment:
Please check also the size of the two remaining entries so that we can be sure that the right mempool was removed.